rfc/20240529-OpenTofu-Specific-Code-Override.md

```diff
@@ -0,0 +1,160 @@
+
+With this solution implemented:  
+1. If I want to use a specific OpenTofu feature in one of my module files, I can create two copies of the file - a `.tf` file and a `.tofu` file with very similar content. The only difference is that the `.tofu` file will contain OpenTofu specific features that are not supported by Terraform. This way, I can use the new features without breaking my module's compatibility with Terraform.
+2. If I want to set a minimum version for Terraform and OpenTofu for my module, I can create a`.tf` file and a `.tofu` file both containing:
+    ```hcl
+    terraform {
+      required_version = ">= x.x.x"
+    }
+    ```
+   In the `.tf` file I'll put my minimum Terraform version, and in the `.tofu` file I'll put my minimum OpenTofu version. This way, I can set different minimum versions for OpenTofu and Terraform in the same module.
+
+#### I’m a module developer/user who wants to write a module/configuration supporting only Tofu
+Today:  
+Unfortunately, I don't have a simple way to do it today. I cannot limit my modules/configurations to be loaded only by OpenTofu.
+
+With this solution implemented:  
+If I include only `.tofu` files in my module/configuration, I can be sure it'll be supported only by OpenTofu. This way, I can use OpenTofu-specific features without worrying that others might run the module/configuration by mistake with Terraform instead of OpenTofu.
+
+#### I’m a TF user who wants to switch to Tofu and start using new Tofu features, without losing the ability to go back to TF
+Today:  
+I can [migrate to OpenTofu](https://opentofu.org/docs/intro/migration/) very easily, and mostly keep my configuration the same way it is with minimal required [code changes in some cases](https://opentofu.org/docs/intro/migration/terraform-1.8/#step-5-code-changes). But for using OpenTofu-specific configuration I still need to change my existing configuration files.
+
+With this solution implemented:  
+I have another option for the migration process. Instead of changing my existing `.tf` configuration files to try OpenTofu-specific features, I can create a new `.tofu` file with the same content as the `.tf` file and start using OpenTofu-specific features in the `.tofu` file. This way, I can experiment with OpenTofu-specific features without changing my existing `.tf` files. If I decide to go back, I can simply delete the `.tofu` file and keep using the `.tf` file.
```

Comment:
This is an RFC talking about possible user stories, not a migration manual for the users. I don't think we should add this caveat here. Tell me if you think otherwise.